doc/tutorial.rst

```diff
@@ -597,4 +597,46 @@ node positions, or write the graph in dot format for further processing.
 
 See :doc:`/reference/drawing` for additional details.
 
+Reading and writing graphs
+--------------------------
+
+.. csv-table:: Supported file formats
+   :header: "Format", "Description"
+   :widths: 20, 40
+
+   "Adjacency list", "Each row 1st column as out-degree"
+   "Edge list", "2 column, source$\rightarrow$ target"
+   "DOT", "Graphviz format"
+   "GEXF", "XML implementation"
+   "GML", "Similar to DOT"
+   "GraphML", "XML implementation"
+   "LEDA", "Between edge list and Pajek"
+   "Pajek", "Edge list + node and edge attr"
+   "SparseGraph6", "Adjacency list variant"
+
+You can read and write NetworkX graphs as Python pickles.
```

Comment:
I'd start with the example you'd like users to follow.  Pickles aren't compatible across versions typically, so perhaps this isn't it.

Reply:
I am not sure what to do here.  I didn't put either the yaml or pickle example here because I thought people should do it.  We used to have functions to write and read yaml and pickle files.  I deprecated them because they didn't seem very useful and were basically one-liners.  Without thinking much more than that, I thought they deserved a mention in the docs at best.  But I am not sure they deserve even that.  Maybe we should just leave them both out of the documentation.  We could just leave the table and the link to the docs.  Thoughts?

@dschult Is there a preferred method for saving a NetworkX graph?

Reply:
People may forget that NX graphs are dicts-of-dicts, so I think a reminder that they can be written to disk is good.  I'd simply put the "best obvious" way at the top.

Reply:
Would yaml be a better format?

Reply:
Yes, I'd think yaml and json are both fine (and approximately equivalent).

Reply:
There is not a preferred method for saving a NetworkX Graph. 
I think historically most of us have used pickle.

Reply:
Understood, but if we had to give an example, what would we choose? I don't think there is any advantage to using pickle over the others.

Reply:
I don't have time to look into this now, so I just moved this to the 2.7 milestone.  I will go back to my original plan of learning more about graph file formats.  Once I understand the issues better, I will finish this section of the tutorial.

I have a few links to resources I found online in the first comment on this PR.  If anyone knows of other resources, please add them.